src/NoisePlethora.cpp

```diff
@@ -0,0 +1,892 @@
+		const int section = programSelector.getMode();
+
+		setAlgorithm(section, algorithmName);
+	}
+
+	void setAlgorithmViaBank(int newBank) {
+		newBank = clamp(newBank, 0, numBanks - 1);
+		const int currentProgram = programSelector.getCurrent().getProgram();
+		// the new bank may not have as many algorithms
+		const int currentProgramInNewBank = clamp(currentProgram, 0, getBankForIndex(newBank).getSize() - 1);
+		const std::string algorithmName = getBankForIndex(newBank).getProgramName(currentProgramInNewBank);
+		const int section = programSelector.getMode();
+
+		setAlgorithm(section, algorithmName);
+	}
+
+	void setAlgorithm(int section, std::string algorithmName) {
+
+		if (section > 1) {
+			return;
+		}
+
+		for (int bank = 0; bank < numBanks; ++bank) {
+			for (int program = 0; program < getBankForIndex(bank).getSize(); ++program) {
+				if (getBankForIndex(bank).getProgramName(program) == algorithmName) {
+					programSelector.setMode(section);
+					programSelector.getCurrent().setBank(bank);
+					programSelector.getCurrent().setProgram(program);
+
+					// update dial
+					params[PROGRAM_PARAM].setValue(programKnobMode == PROGRAM_MODE ? program : bank);
+
+					algorithm[section] = MyFactory::Instance()->Create(algorithmName);
+
+					if (algorithm[section]) {
+						algorithm[section]->init();
+					}
+					else {
+						DEBUG("WARNING: Failed to initialise %s in programSelector", algorithmName.c_str());
+					}
+
+					return;
+				}
+			}
+		}
+
+		DEBUG("WARNING: Didn't find %s in programSelector", algorithmName.c_str());
+	}
+
+	void dataFromJson(json_t* rootJ) override {
+		json_t* bankAJ = json_object_get(rootJ, "algorithmA");
+		if (bankAJ) {
+			setAlgorithm(SECTION_A, json_string_value(bankAJ));
+		}
+
+		json_t* bankBJ = json_object_get(rootJ, "algorithmB");
+		if (bankBJ) {
+			setAlgorithm(SECTION_B, json_string_value(bankBJ));
+		}
+
+		json_t* bypassFiltersJ = json_object_get(rootJ, "bypassFilters");
+		if (bypassFiltersJ) {
+			bypassFilters = json_boolean_value(bypassFiltersJ);
+		}
+
+		json_t* blockDCJ = json_object_get(rootJ, "blockDC");
+		if (blockDCJ) {
+			blockDC = json_boolean_value(blockDCJ);
+		}
+	}
+
+	json_t* dataToJson() override {
+		json_t* rootJ = json_object();
+
+		json_object_set_new(rootJ, "algorithmA", json_string(programSelector.getA().getCurrentProgramName().c_str()));
+		json_object_set_new(rootJ, "algorithmB", json_string(programSelector.getB().getCurrentProgramName().c_str()));
+
+		json_object_set_new(rootJ, "bypassFilters", json_boolean(bypassFilters));
+		json_object_set_new(rootJ, "blockDC", json_boolean(blockDC));
+
+		return rootJ;
+	}
+};
+
+
+struct BefacoTinyKnobSnapPress : BefacoTinyKnobBlack {
+	BefacoTinyKnobSnapPress() {
+		snap = true;
+		minAngle = -0.80 * M_PI;
+		maxAngle = 0.80 * M_PI;
+	}
+
+	// this seems convoluted but I can't see how to achieve the following (say) with onAction:
```

Comment:
If there's an easier way, let me know, but I couldn't see how to get all three...